Files/UserControls/PreviewPane.xaml.cs

```diff
@@ -0,0 +1,241 @@
+        }
+
+        public event PropertyChangedEventHandler PropertyChanged;
+        void RaisePropertyChanged(string name)
+        {
+            if (PropertyChanged != null)
+            {
+                PropertyChanged(this, new PropertyChangedEventArgs(name));
+            }
+        }
+
+        async void LoadPreviewControlAsync(ListedItem item)
+        {
+            PreviewNotAvaliableText.Visibility = Visibility.Collapsed;
+
+            // Folders not supported yet
+            if (item.FileExtension == null)
+            {
+                PreviewNotAvaliableText.Visibility = Visibility.Visible;
+                return;
+            }
+
+            foreach (var extension in AppData.FilePreviewExtensionManager.Extensions)
+            {
+                if (extension.FileExtensions.Contains(item.FileExtension))
+                {
+                    LoadPreviewControlFromExtension(item, extension);
+                    return;
+                }
+            }
+
+            var control = GetBuiltInPreviewControl(item);
+            if (control != null)
+            {
+                PreviewGrid.Children.Add(control);
+                return;
+            }
+
+            control = await TextPreview.TryLoadAsTextAsync(item);
+            if(control != null)
+            {
+                PreviewGrid.Children.Add(control);
+                return;
+            }
+
+            PreviewNotAvaliableText.Visibility = Visibility.Visible;
+        }
+
+        UserControl GetBuiltInPreviewControl(ListedItem item)
+        {
+            var ext = item.FileExtension.ToLower();
+            if (MediaPreview.Extensions.Contains(ext))
+            {
+                return new MediaPreview(item);
+            }
+
+            if (MarkdownPreview.Extensions.Contains(ext))
+            {
+                return new MarkdownPreview(item);
+            }
+
+            if (ImagePreview.Extensions.Contains(ext))
+            {
+                return new ImagePreview(item);
+            }
+
+            if (TextPreview.Extensions.Contains(ext))
+            {
+                return new TextPreview(item);
+            }
+
+            if(PDFPreview.Extensions.Contains(ext))
+            {
+                return new PDFPreview(item);
+
+            }
+
+            if (HtmlPreview.Extensions.Contains(ext))
+            {
+                return new HtmlPreview(item);
+            }
+
+            if (RichTextPreview.Extensions.Contains(ext))
+            {
+                return new RichTextPreview(item);
+            }
+
+            if(CodePreview.Extensions.Contains(ext))
+            {
+                return new CodePreview(item);
+            }
+
+            return null;
+        }
+
+        async void LoadPreviewControlFromExtension(ListedItem item, Helpers.Extension extension)
+        {
+            var file = await StorageFile.GetFileFromPathAsync(item.ItemPath);
+
+            var buffer = await FileIO.ReadBufferAsync(file);
```

Comment:
Should we just pass the filepath to the extension? Loading the whole file could be really big.

Reply:
@gave92 I think the issue with that is then the extensions would have to have the full filesystem access capability.

Reply:
But it's probably a good idea to put a file size limit on that and just send the path.

Reply:
@winston-de right I didn't think of that 🤦‍♂️ There's no way to "share" the access to a StorageFile with an extension is there? The extensions would have to open a file picker

Reply:
Perhaps you can actually do it: https://docs.microsoft.com/en-us/uwp/api/Windows.ApplicationModel.DataTransfer.SharedStorageAccessManager?view=winrt-19041

Reply:
@gave92 Thanks, that's a much better solution! I probably won't be able to implement it for this PR though. For the time being I just put a file size limit.

Reply:
@gave92 I was able to get it working with no issues. Thanks again for the tip!

Reply:
A bit out of scope for this PR, but we should probably use that API when launching files as well.